### PR TITLE
Feat ruby response models

### PR DESCRIPTION
**Pull request by abnegate**
- Add ruby response models
- Enable ruby exceptions tests
- Add ruby 3.0 test

### CODE REVIEW

File: templates/ruby/gemspec.twig [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+  spec.name = '{{spec.title | caseLower | caseSnake}}'
+  spec.version = '{{ sdk.version }}'
+  spec.license = '{{ spec.licenseName }}'
+  spec.summary = '{{ sdk.shortDescription }}'
+  spec.author = '{{ spec.contactName }}'
+  spec.homepage = '{{ spec.contactURL }}'
+  spec.email = '{{ spec.contactEmail }}'
+  spec.files = Dir['lib/**/*.rb']
+
+  spec.add_dependency 'mime-types', '~> 3.3.1'
```
[the remark this answers is not in the document]
Comment (abnegate) [reply templates/ruby/gemspec.twig:12]: Yep, there is some overlap but the two are used with different tools. Gemfile is used by bundler and gemspec by rubygems. Major difference is this way, `mime-types` is defined as a dependency of the Appwrite gem - while the original way defines `mime-types` as a dependency of a bundle that also defines the Appwrite gem.

TLDR;
- Libraries should use `.gemspec`
- Applications should use `Gemfile`

**Comment by abnegate** [comment]
@lohanidamodar Pushed on [dev](https://github.com/appwrite/sdk-for-ruby/tree/dev) 🙂

**Review by christyjacob4 — APPROVED** [review]

**Review by lohanidamodar — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]
